### ISSUE TITLE
Pretrained Model can't load?

### ISSUE THREAD

**Issue by HeConnor** [lab-smile/GRACE#9]
Hello,

I got error when I load pretrained model:

> monai: v0.8.1

```
Error(s) in loading state_dict for UNETR:
	Missing key(s) in state_dict: "vit.patch_embedding.cls_token", "encoder2.blocks.0.1.conv3.conv.weight", "encoder2.blocks.1.1.conv3.conv.weight", "encoder3.blocks.0.1.conv3.conv.weight". 
```

- ！ another question: When T1 images are used as network inputs, it need to be normalized from [0, 255] to [0, 1]; does this mean the maximum value of T1 images is only 255 ? ?
```
parser.add_argument("--a_max_value", type=int, default=255, help="maximum image intensity")
```

Any help is welcome.

Thanks !

**Comment by yuanjustin** [comment]
I believe the author deliberately modified the model, which raises doubts about the authenticity of the data in the article.

**Comment by SkylarStolte** [comment]
Hi HeConnor and yuanjustin, the T1 images that I used were normalized to have a maximum of 255 in advance, which is why the default maximum was 255. Other maximum T1 values should work as long as the normalization to 0-1 is correct.  The loading_state_dict error may be due to data_parallel. What did you set data_parallel equal to with test.sh? Why do you believe that the model was modified? Did you use the version from the github release?

**Comment by monsterYuanguan** [comment]
Dear SkylarStolte. 

Please accept my apologies. The reason I believed that the model had been modified was because many people had already pointed out issues with it, yet the author had not responded, which made it seem like they were avoiding the topic. However, your prompt reply has cleared up my doubts.

That said, regardless of whether the model is downloaded from the release page or the cloud drive link, all versions appear to be missing the following four keys:

- "module.vit.patch_embedding.cls_token"
- "module.encoder2.blocks.0.1.conv3.conv.weight"
- "module.encoder2.blocks.1.1.conv3.conv.weight"
- "module.encoder3.blocks.0.1.conv3.conv.weight"

**Comment by lab-smile** [comment]
I cc Skylar on this email to respond the details.
Ruogu

Sent from my iPhone

On May 30, 2025, at 9:09 PM, yuanjustin ***@***.***> wrote:

﻿
[External Email]
[https://avatars.githubusercontent.com/u/12294912?s=20&v=4]yuanjustin left a comment (lab-smile/GRACE#9)<https://github.com/lab-smile/GRACE/issues/9#issuecomment-2923882243>

Dear SkylarStolte.

Please accept my apologies. The reason I believed that the model had been modified was because many people had already pointed out issues with it, yet the author had not responded, which made it seem like they were avoiding the topic. However, your prompt reply has cleared up my doubts.

That said, regardless of whether the model is downloaded from the release page or the cloud drive link, all versions appear to be missing the following four keys:

  *   "module.vit.patch_embedding.cls_token"
  *   "module.encoder2.blocks.0.1.conv3.conv.weight"
  *   "module.encoder2.blocks.1.1.conv3.conv.weight"
  *   "module.encoder3.blocks.0.1.conv3.conv.weight"

**Comment by HeConnor** [comment]
I appreciate your reply. (life is very hard, so don’t hurt us with the truth ༼ つ ◕_◕ ༽つ)

**Comment by jindalankush28** [comment]
Hi @SkylarStolte  and @lab-smile, 

I can confirm that the issues are still open and waiting for a resolution. I believe the rest of the code is fining and running as expected, Lack of availability of the 4 keys mentioned above is resulting in horrible segmentation. Wondering if you could provide a solution and upload the model file again ? I have tried downloading it from dropbox as well as gdrive.
